### PR TITLE
[MM-56017] Remove prereq-check.sh redantant script from Make

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- Removed the 'prereq-check.sh' script because it redundantly checked node, npm, and go versions only in the make run target. The targets already have a go version check, and node and npm versions are verified in the webapp's engine field.
- Renamed `M1_MAC` variable in Make to `ARM_BASED_MAC`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56017

#### Screenshots
none

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]
This seems fine, @M-ZubairAhmed, but I honestly have no idea what this file does, and it seems to be stale for Go as well? Are we sure we need it?

**Review by isacikgoz — APPROVED** [review]
LGTM but curious on similar question; doesn't `package.json` or lock file provide this information?

File: server/Makefile [hunk 1]
```diff
@@ -623,7 +620,7 @@ run-client-fullmap: client ## Legacy alias to run-client
 
 	cd $(BUILD_WEBAPP_DIR) && $(MAKE) run
 
-run: check-prereqs run-server run-client ## Runs the server and webapp.
+run: run-server run-client ## Runs the server and webapp.
```
Comment (M-ZubairAhmed) [inline server/Makefile:623]: run-server target has go version check already

**Pull request merged**